### PR TITLE
agent is updated according to the karate v0.9.3

### PR DESCRIPTION
**Pull request by ogarashchuk**

### CODE REVIEW

**Comment by HardNorth** [comment]
And a general question. I'm not too familiar with Karate, but our previous implementaion based on `Runner` pattern. We extending Karate's runner and apply it with `@RunWith` annotation, why do you choose different approach?

File: src/test/resources/logback.xml [hunk 1]
```diff
@@ -22,11 +22,20 @@
         <appender-ref ref="RP"/>
     </logger>
 
+    <appender name="FILE" class="ch.qos.logback.core.FileAppender">
```
[the remark this answers is not in the document]
Comment (ogarashchuk) [reply src/test/resources/logback.xml]: fixed

File: src/main/java/com/epam/RPReporter.java [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        rq.setName(parameters.getLaunchName());
+        rq.setStartTime(Calendar.getInstance().getTime());
+        rq.setMode(parameters.getLaunchRunningMode());
+        if (!isNullOrEmpty(parameters.getDescription())) {
+            rq.setDescription(parameters.getDescription());
+        }
+        return rq;
+    }
+
+    synchronized void startFeature(FeatureResult featureResult) {
```
[the remark this answers is not in the document]
Comment (ogarashchuk) [reply src/main/java/com/epam/RPReporter.java]: legacy from previous implementation, removed

File: pom.xml [hunk 3]
```diff
@@ -30,52 +29,45 @@
     <dependencies>
         <dependency>
             <groupId>com.intuit.karate</groupId>
-            <artifactId>karate-apache</artifactId>
-            <version>0.7.0.2</version>
-            <scope>test</scope>
+            <artifactId>karate-core</artifactId>
+            <version>0.9.3</version>
```
[the remark this answers is not in the document]
Comment (ogarashchuk) [reply pom.xml]: the property has been created. Version cannot be updated as customer uses karate 0.9.3

File: src/main/java/com/epam/AgentRunner.java [hunk 4]
```diff
@@ -0,0 +1,175 @@
+package com.epam;
```
[the remark this answers is not in the document]
Comment (ogarashchuk) [reply src/main/java/com/epam/AgentRunner.java]: fixed

File: src/main/java/com/epam/AgentRunner.java [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.util.*;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.function.Consumer;
+
+public class AgentRunner extends Runner {
```
[the remark this answers is not in the document]
Comment (ogarashchuk) [reply src/main/java/com/epam/AgentRunner.java]: agreed, useless extension, removed

File: src/main/java/com/epam/AgentRunner.java [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+
+    private static final Logger logger = LoggerFactory.getLogger(AgentRunner.class);
```
[the remark this answers is not in the document]
Comment (ogarashchuk) [reply src/main/java/com/epam/AgentRunner.java]: fixed

**Comment by ogarashchuk** [comment]
> Please update your merge request a little bit.
> 
> And a general question. I'm not too familiar with Karate, but our previous implementaion based on Runner pattern. We extending Karate's runner and apply it with @RunWith annotation, why do you choose different approach?

@RunWith annotation doesn't work when used parallel test run. As Karate framework is for API testing then in the most of the cases will be used parallel test run. With used approach number of threads can be set to 1 and it will bee equal to @RunWith annotation.

**Comment by evgenytkachenko** [comment]
can it be merged @HardNorth @DzmitryHumianiuk  ?
I built it locally and made it work with a couple of changes

**Comment by HardNorth** [comment]
@evgenytkachenko The issue with this code is that it's not binary compatible with previous version. We are going to create a separate repository for the agent v0.9.3.

**Comment by evgenytkachenko** [comment]
@ogarashchuk did you make it work with 0.9.5?
Also I see an issue with the order of steps in RP when I use your implementation.

**Comment by evgenytkachenko** [comment]
@ogarashchuk the test from your example does not send karate debug-level log to RP, it workes in the older version

**Comment by ogarashchuk** [comment]
> @ogarashchuk the test from your example does not send karate debug-level log to RP, it workes in the older version

@evgenytkachenko, I offer to close this PR as we changed the approach for customer how to work with this version of Karate Framework and Report Portal

**Comment by evgenytkachenko** [comment]
@ogarashchuk are you going to create another one PR to integrate Report Portal v5 with Karate 0.9.5 ?

**Comment by ogarashchuk** [comment]
> @ogarashchuk are you going to create another one PR to integrate Report Portal v5 with Karate 0.9.5 ?

@evgenytkachenko, we are still using Karate 0.9.3 but with changed approach to RP integration. Customer wants to have results imported to RP after test run (not in run time) in JUnit format. The convertor for results from json to xml (junit) and importer of the results to RP was implemented by one of epamers and uploaded to the public access.

**Comment by evgenytkachenko** [comment]
@ogarashchuk so you just grab an xml file from surefire-reports folder compress it and then import to the RP using bash script and RP API ?

**Comment by ogarashchuk** [comment]
> @ogarashchuk so you just grab an xml file from surefire-reports folder compress it and then import to the RP using bash script and RP API ?

no exactly, json files that generated by Karate to target/reports (folder can be specified in test runner) converted to xml files in junit format and imported to the RP using RP API

**Comment by evgenytkachenko** [comment]
@ogarashchuk sounds good
you aggregate all json files and convert them into Junit xml file?
can you share this solution ?

**Comment by varunpatil03** [comment]
Hi All,
Sorry to cut this chain. I am trying to integrate my reportportal to karate projects in my company. I want to understand if this is supported for Junit Runner as of today? I am using Karate 0.9.4 version.

**Comment by ramrasan** [comment]
Hi All, 
2 questions:
1.When are you expecting to release the new support
2. Will it support also Karate 0.95

**Comment by ogarashchuk** [comment]
closed due to outdated version of the Karate Framework

**Pull request closed without merging**